src/Communication/Session.php

```diff
@@ -136,5 +136,6 @@ public function destroy(): void
         $this->emit('destroyed');
         $this->connection = null;
         $this->removeAllListeners();
+        $this->destroyed = true;
```

Comment:
We maybe wanna do that before calling removeAllListeners?

Reply:
maybe but i don't know if it have any difference as removeAllListener doesn't call any other code